sipXproxy/lib/authplugins/EnforceAuthRules.cpp

```diff
@@ -269,16 +269,26 @@ EnforceAuthRules::authorizeAndModify(const UtlString& id,    /**< The authentica
             }
             else
             {
-               result = DENY;
-               Os::Logger::instance().log(FAC_AUTH, PRI_WARNING,
-                             "EnforceAuthRules[%s]::authorizeAndModify "
-                             " call '%s' requires '%s'",
-                             mInstanceName.data(), callId.data(), unmatchedPermissions.data()
-                             );
-               // since the user is at least a valid user, help them debug the configuration
-               // by telling them what permissions would allow this request.
-               reason.append("Requires ");
-               reason.append(unmatchedPermissions);
+              if (!unmatchedPermissions.isNull())
+              {
+                result = DENY;
+                Os::Logger::instance().log(FAC_AUTH, PRI_WARNING,
+                              "EnforceAuthRules[%s]::authorizeAndModify "
+                              " call '%s' requires '%s'",
+                              mInstanceName.data(), callId.data(), unmatchedPermissions.data()
+                              );
+                // since the user is at least a valid user, help them debug the configuration
+                // by telling them what permissions would allow this request.
+                reason.append("Requires ");
+                reason.append(unmatchedPermissions);
+              }
+              else
+              {
+                result = ALLOW;
```

Comment:
Guys,

isAuthorized returned false here so ALLOW is very much suspect.  Checking the reason where unmatchedPermission is not set but isAuthorized returns false is when findIdentity returns false.    So this must be denied.   The bug here is not rejecting the call but rather unmatchPermission being NULL.   If it was NULL because there is no user found, the logs should be clear about that.   Nothing else should change.